ocaml/xapi/xapi_host.ml

```diff
@@ -1413,19 +1411,18 @@ let reset_server_certificate ~__context ~host =
   in
   let dns_names = Networking_info.dns_names () in
   let ips = [ip] in
-  Gencertlib.Selfcert.host ~name ~dns_names ~ips xapi_ssl_pem ;
-  (* Reset stunnel to try to restablish TLS connections *)
-  Xapi_mgmt_iface.reconfigure_stunnel ~__context ;
-  (* Delete records of the server certificate in this host *)
-  let expr =
-    Db_filter_types.(Eq (Field "host", Literal (Ref.string_of self)))
-  in
-  Db.Certificate.get_refs_where ~__context ~expr
-  |> List.iter (fun self -> Db.Certificate.destroy ~__context ~self)
-
-let emergency_reset_server_certificate ~__context =
-  let host = Helpers.get_localhost ~__context in
-  reset_server_certificate ~__context ~host
+  Gencertlib.Selfcert.host ~name ~dns_names ~ips xapi_ssl_pem
+
+let reset_server_certificate ~__context ~host =
+  let dbg = Context.string_of_task __context in
+  let write_cert_fs () = _new_host_cert ~dbg in
+  replace_host_certificate ~__context ~host ~complete_task:false write_cert_fs
```

Comment:
Thinking of it, now that resetting the server certificate is not local, this should be set to 'true' and add a tolerate_connection_loss  in the corresponding function in message_forwarding.

Reply:
This requires removing the ```Session`` flag from the call in datamodel_host.ml - I'm not sure why that flag was there in the first place, possibly because it was copied from `emergency_reset_server_certificate`